### PR TITLE
Move pkg/api/ref.go to a subpackage

### PR DESCRIPTION
**Pull request by caesarxuchao**
First commit is #44296. (unfortunately, removing that commit results in conflicts)

This PR moves the pkg/api/ref.go to its own subpackage. It's mostly a mechanic move.

I'll send a few more PRs to make the k8s.io/kubernetes/pkg/api package only contains the code we want in the k8s.io/api repo, then we can run a [script](https://github.com/kubernetes/kubernetes/pull/41747/commits/a0015fd1be6c6a239fc40cc3c04ba5ed7b1ab22e#diff-7a2fbb4371972350ee414c6b88aee1c8) to cut the new repo.

### CODE REVIEW

File: pkg/api/methods.go [hunk 1]
```diff
@@ -44,3 +48,12 @@ func (t *Taint) ToString() string {
 	}
 	return fmt.Sprintf("%v=%v:%v", t.Key, t.Value, t.Effect)
 }
+
+func (obj *ObjectReference) SetGroupVersionKind(gvk schema.GroupVersionKind) {
+	obj.APIVersion, obj.Kind = gvk.ToAPIVersionAndKind()
+}
```
Comment (lavalamp) [inline pkg/api/methods.go]: nit: missing blank line

I would slightly prefer not to make a methods.go file which is a dumping ground for methods, what if we kept a ref.go here with just these methods?

Comment (lavalamp) [inline pkg/api/methods.go]: (and maybe change name of existing one to taints.go or some such)

**Comment by lavalamp** [comment]
Can fix my comment in a follow-up if that's easier. Looks like this could be squashed a bit though.

File: pkg/api/methods.go (hunk 1 above)
Comment (caesarxuchao) [inline pkg/api/methods.go]: Ok. I'll address this and squash.

**Comment by caesarxuchao** [comment]
@lavalamp PTAL. Thanks.

**Comment by caesarxuchao** [comment]
@k8s-bot cvm gce e2e test this

**Comment by caesarxuchao** [comment]
Closing this. I'll squash it with #44302.

**Pull request merged**